clang/lib/Sema/SemaExpr.cpp

```diff
@@ -228,7 +228,8 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
                              ObjCInterfaceDecl *ClassReceiver,
                              bool SkipTrailingRequiresClause) {
   if (getLangOpts().SYCLIsDevice) {
-    if (auto VD = dyn_cast<VarDecl>(D)) {
+    if (auto VD = dyn_cast<VarDecl>(D);
+        VD && !VD->getType()->isDependentType()) {
```

Comment:
Nice! I forgot that if-with-separate-initializer-and-condition was in C++17 and that we can use it now! Love it!